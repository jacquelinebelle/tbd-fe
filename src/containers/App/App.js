import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import ResultsPage from '../ResultsPage';
import './App.scss';
import Form from '../Form/';
import Header from '../Header';
import JobDetailPage from '../../components/JobDetailPage';
import CitiesPage from '../../components/CitiesPage/CitiesPage';
// import JobListingContainer from '../../components/JobListingContainer/JobListingContainer';

export class App extends Component {
constructor() {
  super()
  this.state={
    
  }
}

  render() {
    return (
      <Switch>
        <div className="App">
          <img alt="aerial view of a cirty with scyscrapers" src='https://wallpapercave.com/wp/wp3594884.jpg' className='result-background'/>
          <Header />
          <Route 
            exact path='/' 
            render={ () => <Form/> } 
          />
          <Route 
            exact path='/results' 
            component={ResultsPage} 
          />
          <Route
            path="/cities"
            component={CitiesPage}
            />
          <Route exact path={`/job/:id`} render={
                (id) => (<JobDetailPage id={id.location.pathname} history={id.history} />)
            }
            />
        </div>
      </Switch>
    )
  }
}

export default App;
