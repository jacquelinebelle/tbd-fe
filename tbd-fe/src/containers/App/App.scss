@import-normalize;
@import '../../assets/variables.scss';

.App {
    box-sizing: border-box;
    color: $font-color;
    font-family: $primary-font;
    font-weight: $primary-weight;
    button {
        font-weight: $primary-weight;
    }
}

h1 {
    font-weight: $large-text-weight;
    letter-spacing: 2px;
    margin: 20px 0;
    margin: 20px auto;
    width: 30px;
}
