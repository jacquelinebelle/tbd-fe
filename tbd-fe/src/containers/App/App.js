import React, { Component } from 'react';
import { Route, Switch, Link } from 'react-router-dom';
import ResultsPage from '../ResultsPage';
import './App.scss';
import Form from '../Form/';
import Header from '../Header';
// import JobListingContainer from '../../components/JobListingContainer/JobListingContainer';

export class App extends Component {
constructor() {
  super()
  this.state={
    
  }
}

  render() {
    return (
      <Switch>
        <div className="App">
          <img src='https://wallpapercave.com/wp/wp3594884.jpg' className='result-background'/>
          <Header />
          <Route 
            exact path='/' 
            render={ () => <Form/> } 
          />
          <Route 
            exact path='/results' 
            component={ResultsPage} 
          />

        </div>
      </Switch>
    )
  }
}

export default App;
