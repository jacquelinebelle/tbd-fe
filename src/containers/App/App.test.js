import React from 'react';
import {shallow} from 'enzyme';
import App from './App';


describe('App', () => {
  let wrapper;

  beforeEach(function() {
    wrapper = shallow(
      <App />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})

