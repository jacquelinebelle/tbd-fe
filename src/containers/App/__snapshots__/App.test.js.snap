// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<Switch>
  <div
    className="App"
  >
    <img
      alt="aerial view of a cirty with scyscrapers"
      className="result-background"
      src="https://wallpapercave.com/wp/wp3594884.jpg"
    />
    <Header />
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(ResultsPage)",
          "type": [Function],
        }
      }
      exact={true}
      path="/results"
    />
    <Route
      component={[Function]}
      path="/cities"
    />
    <Route
      exact={true}
      path="/job/:id"
      render={[Function]}
    />
  </div>
</Switch>
`;
